src/oop/user_defined_classes/medium/q7/MainTest.java

```diff
@@ -0,0 +1,94 @@
+        assertTrue(classInstance.hasField(var2, int.class), modifiedClassMessage);
+        assertTrue(classInstance.hasModifier(var2, "private"), modifiedClassMessage);
+        Class<?>[] classArguments = {int.class};
+        assertTrue(classInstance.hasConstructor(classArguments), modifiedClassMessage);
+        assertTrue(classInstance.hasModifier(classArguments, "public"), modifiedClassMessage);
+        Object[][] tests = new Object[][]{
+                {120},
+                {60},
+                {30}
+        };
+        for (int num = 0; num < tests.length; num++) {
+            int speed = (int) tests[num][0];
+            Object[][] arguments = {
+                    {speed, int.class}
+            };
+            Object instance = classInstance.createInstance(arguments);
+            _assertEquals(speed, classInstance.getFieldValue(instance, var1), modifiedClassMessage);
+            _assertEquals(initialFee, classInstance.getFieldValue(instance, var2), modifiedClassMessage);
+        }
+
+
+    }
+
+    private static Stream<Arguments> methodInputProvider() {
+        return Stream.of(
+                Arguments.of(120, 40, 0),
+                Arguments.of(60, 70, initialFee),
+                Arguments.of(30, 40, initialFee),
+                Arguments.of(100, 100, 0),
+                Arguments.of(30, 31, initialFee),
+                Arguments.of(100, 99, 0)
+        );
+    }
+
+    @Test
+    public void issueTicketIsDefinedCorrectly() {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(classInstance.hasMethod(methodName, new Class<?>[]{int.class}), String.format(incorrectMethodDefinition, methodName, className));
+        assertTrue(classInstance.hasModifier(methodName, new Class<?>[]{int.class}, "public"), String.format(incorrectModifierMessage, methodName, className));
+        assertTrue(classInstance.hasReturnType(methodName, new Class<?>[]{int.class}, int.class), String.format(incorrectReturnType, methodName, className));
+    }
+
+    @ParameterizedTest
+    @MethodSource("methodInputProvider")
+    public void highwayCorrectIssueTicketMethod(int speedLim, int current, int ans) throws Throwable {
+        Object[][] arguments1 = {
+                {speedLim, int.class}
+        };
+        Object[][] arguments2 = {
+                {current, int.class}
+        };
+        Object instance = classInstance.createInstance(arguments1);
+        Object output = classInstance.callMethod(methodName, arguments2, (new String[]{"public"}), instance);
+        assertTrue(classInstance.hasMethod(methodName, new Class<?>[]{int.class}, int.class, new String[]{"public"}),
```

Comment:
Please separate this check out so the student will receive separate feedback on the return type and visibility modifier, and make sure it comes before any calls to the constructor or calling the method.